src/main/telemetry/srxl.c

```diff
@@ -83,24 +84,33 @@
 #define SRXL_FRAMETYPE_GPS_STAT     0x17
 
 static bool srxlTelemetryEnabled;
+static bool srxl2 = false;
 static uint8_t srxlFrame[SRXL_FRAME_SIZE_MAX];
 
 static void srxlInitializeFrame(sbuf_t *dst)
 {
-    dst->ptr = srxlFrame;
-    dst->end = ARRAYEND(srxlFrame);
+    if (srxl2) {
+      srxl2InitializeFrame(dst);
+    } else {
+        dst->ptr = srxlFrame;
+        dst->end = ARRAYEND(srxlFrame);
 
-    sbufWriteU8(dst, SRXL_ADDRESS_FIRST);
-    sbufWriteU8(dst, SRXL_ADDRESS_SECOND);
-    sbufWriteU8(dst, SRXL_PACKET_LENGTH);
+        sbufWriteU8(dst, SRXL_ADDRESS_FIRST);
+        sbufWriteU8(dst, SRXL_ADDRESS_SECOND);
+        sbufWriteU8(dst, SRXL_PACKET_LENGTH);
+    }
 }
 
 static void srxlFinalize(sbuf_t *dst)
 {
-    crc16_ccitt_sbuf_append(dst, &srxlFrame[3]); // start at byte 3, since CRC does not include device address and packet length
-    sbufSwitchToReader(dst, srxlFrame);
-    // write the telemetry frame to the receiver.
-    srxlRxWriteTelemetryData(sbufPtr(dst), sbufBytesRemaining(dst));
+    if (srxl2) {
+      srxl2FinalizeFrame(dst);
```

Comment:
:space_invader:

(Indentation in existing code in this file seems to be dodgy. I think it's best if we leave it as it is for this pull request to facilitate reviewing, and then open a new pull request with indentation fixes only once this has been merged.)